cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java

```diff
@@ -0,0 +1,380 @@
+                }
+            } else if (statusCode == HttpURLConnection.HTTP_NOT_FOUND) {
+                try (BufferedReader reader = new BufferedReader(
+                        new InputStreamReader(conn.getInputStream(), Charset.defaultCharset()))) {
+                    Error errorJsonSchema = new Gson().fromJson(reader, Error.class);
+                    if (errorJsonSchema.getMessage().contains("package not found:")) {
+                        throw new NoPackageException(errorJsonSchema.getMessage());
+                    } else {
+                        throw createLauncherException(
+                                ERR_CANNOT_CONNECT + pkg + ". reason: " + errorJsonSchema.getMessage());
+                    }
+                } catch (IOException e) {
+                    throw ErrorUtil.createCommandException(e.getMessage());
+                }
+            } else if (statusCode == HttpURLConnection.HTTP_BAD_REQUEST) {
+                try (BufferedReader errorStream = new BufferedReader(
+                        new InputStreamReader(conn.getInputStream(), Charset.defaultCharset()))) {
+                    Error errorJsonSchema = new Gson().fromJson(errorStream, Error.class);
+
+                    if (errorJsonSchema.getMessage() != null && !"".equals(errorJsonSchema.getMessage())) {
+                        throw new CommandException(errorJsonSchema.getMessage());
+                    } else {
+                        throw createLauncherException(ERR_CANNOT_CONNECT + pkg + ". reason:" + errorStream.lines()
+                                .collect(Collectors.joining("\n")));
+                    }
+                } catch (IOException e) {
+                    throw ErrorUtil.createCommandException(e.getMessage());
+                }
+            } else {
+                throw createLauncherException(ERR_CANNOT_CONNECT + pkg + ".");
+            }
+        } finally {
+            conn.disconnect();
+            Authenticator.setDefault(null);
+        }
+    }
+
+    /**
+     * Pushing a package to registry.
+     */
+    public void pushPackage(Path baloPath) {
+        final int NO_OF_BYTES = 64;
+        final int BUFFER_SIZE = 1024 * NO_OF_BYTES;
+
+        initializeSsl();
+        HttpURLConnection conn = createHttpUrlConnection(PACKAGES);
+        conn.setInstanceFollowRedirects(false);
+        setRequestMethod(conn, Utils.RequestMethod.POST);
+
+        // Load balo project
+        BaloProject baloProject = BaloProject.loadProject(baloPath);
+        String org = baloProject.currentPackage().packageDescriptor().org().toString();
+        String name = baloProject.currentPackage().packageDescriptor().name().toString();
+        String version = baloProject.currentPackage().packageDescriptor().version().toString();
+
+        // Set headers
+        conn.setRequestProperty(AUTHORIZATION, "Bearer " + this.accessToken);
+        conn.setRequestProperty(PUSH_ORGANIZATION, org);
```

Comment:
This header is not needed for central v2 anymore

Reply:
removed